### PR TITLE
6938/feature/google analytics html for facet

### LINKED ISSUE #6938: Add Google analytics tracking to search facets
We want to know how much these are used; these are expensive to compute, so if any are not used we should remove them or make them lazy. And we should promote the most used ones to the top.

Eg update `<a href="/search?q=the+alchemist&amp;mode=everything&amp;author_key=OL112771A" title="Filter results for Ben Jonson">Ben Jonson</a>` to `<a href="/search?q=the+alchemist&amp;mode=everything&amp;author_key=OL112771A" title="Filter results for Ben Jonson" data-ol-link-track="SearchFacet|Author">Ben Jonson</a>`

So the following events:

- `SearchFacet|(Ebook|Author|Subjects|People|Places|Times|FirstPublished|Publisher|Language` for when someone clicks one of the facets

### PR DESCRIPTION
**Pull request by yoonmgyg**
Closes #6938

Google analytics attributes are added to search facets, allowing for the tracking of their popularity when clicked.

### Technical

work_search.html now contains the data-ol-link-track attribute when headers are clicked, which uses the add_track function to get the value.

### Testing

I'm not entirely sure how it can be verified, but it can be tested by using the search function in the website and clicking on one of the facets provided for the search.

### Stakeholders

@cdrini

### CODE REVIEW

**Comment by cclauss** [comment]
I doubt that webpy will support match / case syntax in html.

**Comment by cclauss** [comment]
Do we NEED match / case or could this be done with normal dict lookup?
```python
    def add_track(key: str) -> str:
        """
        KeyError will be raised if key is not in the following dict.
        """
        facets = {
            "has_fulltext": "Ebook",
            "author_key": "Author",
            "subject_facet": "Subjects",
            "person_facet": "People",
            "place_facet": "Places",
            "time_facet": "Times",
            "first_publish_year": "FirstPublished",
            "publisher_facet": "Publisher",
            "language_facet": "Language",
        }
        return f"SearchFacet|{facets[key]}"
```

**Comment by yoonmgyg** [comment]
Most likely not, I would agree that your version looks much better.

**Comment by cdrini** [comment]
Thanks @yoonmgyg ! I agree with @cclauss ' suggestion :) Could you please update your solution to use that approach?

File: openlibrary/templates/work_search.html [hunk 1]
```diff
@@ -13,29 +13,24 @@
...
-            case "first_publish_year":
-                facet = "FirstPublished"
-            case "publisher_facet":
-                facet = "Publisher"
-            case "language_facet":
-                facet = "Language"
-        return "SearchFacet|" + facet
+            return changequery(page=None,**{k:None}
+)
+    def add_track(key: str) -> str:
```
Comment (cdrini) [inline openlibrary/templates/work_search.html:18]: Sigh it seems like web.py templating also doesn't support types!
```suggestion
    def add_track(key):
```

Comment (cdrini) [reply openlibrary/templates/work_search.html:18]: Actually it somehow does!

**Review by cdrini — APPROVED** [review]
Lgtm! Confirmed the tags now appear on testing.openlibrary.org . Thank you @yoonmgyg !

**Pull request merged**